### PR TITLE
Use default overlap=2988

### PR DESCRIPTION
**Pull request by dkurt**
on master:
```
$ bonito basecaller dna_r9.4.1 reads/test1 --device cpu
AAACAACACATACACACACACACACACACACACACACACACTCTCTCAGTAAATGTGCAGGAAGAAGAGAGACACTATCAATCGAAAACGTATGTAGCTCGAAATTAGGGAGGCAGTAAGTGGGAGGGAGATCTATAAAAGGAAGAAAGAACAGGAAGTTATCCAAGAGAGAGAGATGTATATTGCTCGCGTGCAGAGAGAGAGAGTAGTACATATATATATATATATATATATATATGTGTGTGTGTGTATTTTTAAATCATCGGGTGGAAGCCGGAATAAAAAGCAGGAGATCCCTTTGAAGTAAGTCTCATGATCCATTGCGGGGGTCGTATGCGTGCTGTGGATTAATATAAATGGGTGATCAAAAAAAATTTAGTAAAGGTGAGGCAGCATCATGTATCTCGAGTCCAAAGAAATAATCAAGAGAGAGAGAGATGATAGATAGATCTTGGAGGAGTAAATGAATAATCAAAGA

$ bonito basecaller dna_r9.4.1 reads/test1 --device cpu --chunksize 3600
GGAGAATAAAAAGTCTTGGAGAGAGTAGAAGTGAAGAATCATATCAGAAGAGAAAA

$ bonito basecaller dna_r9.4.1 reads/test1 --device cpu --chunksize 6000
TGAAGGCAGCGATCATGTAATCTCAAGTCCAAAGAAATAATCAAGAGAGAGAGAGATGATAGAAGTAGTTCTTGGAGAGAGATAAATGAAGAATCATATCAGAAGAGAAAA
```

proposed (93% of letters are correct and difference only in last chunk):
```
$ bonito basecaller dna_r9.4.1 reads/test1 --device cpu
AAACAACACATACACACACACACACACACACACACACACACTCTCTCAGTAAATGTGCAGGAAGAAGAGAGACACTATCAATCGAAAACGTATGTAGCTCGAAATTAGGGAGGCAGTAAGTGGGAGGGAGATCTATAAAAGGAAGAAAGAACAGGAAGTTATCCAAGAGAGAGAGATGTATATTGCTCGCGTGCAGAGAGAGAGAGTAGTACATATATATATATATATATATATATATGTGTGTGTGTGTATTTTTAAATCATCGGGTGGAAGCCGGAATAAAAAGCAGGAGATCCCTTTGAAGTAAGTCTCATGATCCATTGCGGGGGTCGTATGCGTGCTGTGGATTAATATAAATGGGTGATCAAAAAAAATTTAGTAAAGGTGAGGCAGCATCATGTATCTCGAGTCCAAAGAAATAATCAAGAGAGAGAGAGATGATAGATAGATCTTGGAGGAGTAAATGAATAATCAAAGA

$ bonito basecaller dna_r9.4.1 reads/test1 --device cpu --chunksize 3600
AAACAACACATACACACACACACACACACACACACACACACTCTCTCAGTAAATGTGCAGGAAGAAGAGAGACACTATCAATCGAAAACGTATGTAGCTCGAAATTAGGGAGGCAGTAAGTGGGAGGGAGATCTATAAAAGGAAGAAAGAACAGGAAGTTATCCAAGAGAGAGAGATGTATATTGCTCGCGTGCAGAGAGAGAGAGTAGTACATATATATATATATATATATATATATGTGTGTGTGTGTATTTTTAAATCATCGGGTGGAAGCCGGAATAAAAAGCAGGAGATCCCTTTGAAGTAAGTCTCATGATCCATTGCGGGGGTCGTATGCGTGCTGTGGATTAATATAAATGGGTGATCAAAAAAAATTTAGTAAAGGTGAGGCAGCATCATGTATCTCGAGTCCAAAGAAATAATCAAGAGAGAGAGAGATGATAGAAGTAGTTCTTGGAGAGAGTGAATGAAAATCTATTCAGAAAGAAAA

$ bonito basecaller dna_r9.4.1 reads/test1 --device cpu --chunksize 6000
AAACAACACATACACACACACACACACACACACACACACACTCTCTCAGTAAATGTGCAGGAAGAAGAGAGACACTATCAATCGAAAACGTATGTAGCTCGAAATTAGGGAGGCAGTAAGTGGGAGGGAGATCTATAAAAGGAAGAAAGAACAGGAAGTTATCCAAGAGAGAGAGATGTATATTGCTCGCGTGCAGAGAGAGAGAGTAGTACATATATATATATATATATATATATATGTGTGTGTGTGTATTTTTAAATCATCGGGTGGAAGCCGGAATAAAAAGCAGGAGATCCCTTTGAAGTAAGTCTCATGATCCATTGCGGGGGTCGTATGCGTGCTGTGGATTAATATAAATGGGTGATCAAAAAAAATTTAGTAAAGGTGAGGCAGCATCATGTATCTCGAGTCCAAAGAAATAATCAAGAGAGAGAGAGATGATAGAAGTAGTTCTTGGAGAGAGATAAATGAAGAATCATATCAGAAGAGAA
```

`2988 = 498 * 6` where 498 is empirically determined number of values which are affected by convolution layers padding.

<details>
<summary>determine_pads.py</summary>

```python
import torch
import numpy as np
from bonito.util import load_model

model = load_model('dna_r9.4.1', 'cpu', weights=0)
model.eval()

inp = np.ones([1, 1, 3600], dtype=np.float32) * 100
out = model(torch.Tensor(inp)).detach().numpy().reshape(-1, 5)
ref_line = out[out.shape[0] // 2]

num = 0
for i, line in enumerate(out):
    if np.max(np.abs(ref_line - line)) == 0.0:
        print(i)
        break
```

</details>

### CODE REVIEW

**Comment by iiSeymour** [comment]
I've been looking at the impact on various chunk size and overlap combinations on some validation sets and found chunks of 3600 samples with an overlap 900 is a good combination when considering speed and accuracy.

File: bonito/basecaller.py [hunk 1]
```diff
@@ -86,8 +92,8 @@ def argparser():
     parser.add_argument("--weights", default="0", type=str)
     parser.add_argument("--beamsize", default=5, type=int)
     parser.add_argument("--batchsize", default=1, type=int)
-    parser.add_argument("--chunksize", default=0, type=int)
-    parser.add_argument("--overlap", default=0, type=int)
+    parser.add_argument("--chunksize", default=0, type=check_chunksize)
+    parser.add_argument("--overlap", default=2988, type=int)
```
Comment (iiSeymour) [inline bonito/basecaller.py:96]: As the chunk size and the overlap are tied to a models stride I think the best place to define them is in the models `config.toml`.

File: bonito/basecaller.py [hunk 2]
```diff
@@ -74,6 +74,12 @@ def main(args):
 
 
 def argparser():
+    def check_chunksize(chunksize):
+        x = int(chunksize)
+        if x != 3 * (x // 3):
```
Comment (iiSeymour) [inline bonito/basecaller.py:79]: I would use modulo here - `if int(chunksize) % 3:`

Comment (dkurt) [reply bonito/basecaller.py]: Sure, replaced, thanks!

File: bonito/basecaller.py (hunk 1 above)
Comment (dkurt) [reply bonito/basecaller.py]: Good point. Proposed to use
```
[inference]
overlap = 2988
```
section for default overlap
